### PR TITLE
add Assets.uploadData() and Assets.downloadData()

### PR DESCRIPTION
**Pull request by ZappoMan**

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 1bbfc39845b0a6fac7160575ffc188a053fc1552 available at: https://gist.github.com/8263b1bdb3e556dd8e23<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6439.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6439.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: f80a765a296caa309f9484f07c12ed3de5f8985e available at: https://gist.github.com/8263b1bdb3e556dd8e23<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6439.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6439.dmg

File: libraries/networking/src/AssetClient.cpp [hunk 1]
```diff
@@ -365,3 +365,65 @@ void AssetClient::handleNodeKilled(SharedNodePointer node) {
...
+    }
+
+    auto assetClient = DependencyManager::get<AssetClient>();
+    auto assetRequest = assetClient->createRequest(hash, extension);
+
+    if (!assetRequest) {
+        return;
+    }
+
+    _pendingRequests << assetRequest;
```
Comment (jherico) [inline libraries/networking/src/AssetClient.cpp]: Not thread safe.

File: libraries/networking/src/AssetClient.cpp [hunk 2]
```diff
@@ -365,3 +365,65 @@ void AssetClient::handleNodeKilled(SharedNodePointer node) {
...
+        if (request->getError() == AssetRequest::Error::NoError) {
+            if (callback.isFunction()) {
+                QString data = QString::fromUtf8(request->getData());
+                QScriptValueList args { data };
+                callback.call(QScriptValue(), args);
+            }
+        }
+
+        request->deleteLater();
+        _pendingRequests.remove(request);
```
Comment (jherico) [inline libraries/networking/src/AssetClient.cpp]: not thread safe

**Pull request merged**